src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -304,6 +332,35 @@ private static ISymbol TryGetSymbol(NameSyntax name, SymbolInfo symbolInfo, Sema
             return symbolInfo.Symbol;
         }
 
+        private bool TryClassifyAsNamespace(
+            NameSyntax name,
+            SymbolInfo symbolInfo,
+            ArrayBuilder<ClassifiedSpan> result)
+        {
+            if (name is IdentifierNameSyntax
+                && symbolInfo.Symbol == null
+                && IsNamespaceName(name))
+            {
+                result.Add(new ClassifiedSpan(name.Span, ClassificationTypeNames.NamespaceName));
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool IsNamespaceName(NameSyntax name)
```

Comment:
i would prefer this named more like IsInNamespaceDeclarationOrUsingDirectiveName